src/libPMacc/include/nvidia/atomic.hpp

```diff
@@ -34,40 +36,113 @@ namespace PMacc
 namespace nvidia
 {
 
+    namespace detail {
+
+        template<typename T_Type, bool T_isKepler>
+        struct AtomicAllInc
+        {
+            DINLINE T_Type
+            operator()(T_Type* ptr)
+            {
+                return atomicAdd(ptr, 1);
+            }
+        };
+
+#if PMACC_CUDA_ARCH >= 300
+       /**
+         * Trait that returns whether an optimized version of AtomicAllInc
+         * exists for Kepler architectures (and up)
+         */
+        template<typename T>
+        struct AtomicAllIncIsOptimized
+        {
+            enum{
+                value = boost::is_same<T,  int32_t>::value ||
```

Comment:
`int32_t` and co are not defined in the [atomicAdd interface of cuda](http://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html)
Please use the types
```
unsigned int
unsigned long long int
...
```